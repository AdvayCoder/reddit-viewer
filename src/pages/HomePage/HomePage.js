import React from "react";
import './HomePage.css';
import { Center, Flex, Text } from "@chakra-ui/react";
import Posts from "../../features/posts/Posts";
import { selectHasError } from "../../features/posts/postsSlice";
import { useSelector } from "react-redux";


const HomePage = () => {

    const hasError = useSelector(selectHasError);

    return (
        //! there may be an issue with the bg later
        <Flex pt='100px' flexDir='column' w='100vw' h='100vh' overflowX='hidden'>
            {!hasError ? (
                <Posts />
            ) : (
                <Center mt='300px'>
                    <Text fontSize='3rem' fontWeight={700}>Oh no! There was an error loading the posts!</Text>
                </Center>
            )}
        </Flex>
    
    )
}

export default HomePage;