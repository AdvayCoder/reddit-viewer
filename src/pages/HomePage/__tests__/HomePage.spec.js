import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import HomePage from '../HomePage';

configure({ adapter: new Adapter() })

describe('<HomePage />', () => {

    it('renders properly', () => {
    const wrapper = shallow(
        <HomePage />
    )

    expect(wrapper).toMatchSnapshot();
    })

    // it('calls componentDidMount', () => {
    //     jest.spyOn(App.prototype, 'componentDidMount')
    //     const wrapper = shallow(<App />)
    //     expect(App.prototype.componentDidMount.mock.calls.length).toBe(1)
    // })
});