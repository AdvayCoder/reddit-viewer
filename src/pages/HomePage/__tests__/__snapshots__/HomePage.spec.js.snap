// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`< /> renders properly 1`] = `ShallowWrapper {}`;

exports[`<> /> renders properly 1`] = `ShallowWrapper {}`;

exports[`<App /> renders properly 1`] = `ShallowWrapper {}`;

exports[`<HomePa /> renders properly 1`] = `ShallowWrapper {}`;

exports[`<HomePage /> renders properly 1`] = `ShallowWrapper {}`;
